src/lib.rs

```diff
@@ -110,11 +110,12 @@ mod macros;
 
 pub mod io;
 pub mod nr;
+pub mod debug;
 
 /// Performs a semihosting operation
 #[inline(always)]
 #[cfg(target_arch = "arm")]
-pub unsafe fn syscall<T>(mut nr: usize, arg: &T) -> usize {
+pub unsafe fn syscall<T: Sized>(mut nr: usize, arg: T) -> usize {
```

Comment:
Well, maybe it is overloaded too much. There are two kinds of calls, the ones that take pointer and the ones that take number. I can create a separate method for the latter.